### ISSUE TITLE
Error in tf.nn.softmax_cross_entropy_with_logits_v2 due to tensor shape change

### ISSUE THREAD

**Issue by shkdidrlf** [tensorflow/tensorflow#24981]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Cent OS 6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): installed from source
- TensorFlow version (use command below):  tensorflow-1.8.0
- Python version: python 3.0
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: cuda 9.0
- GPU model and memory: Tesla V100 (16GB)

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**
Error in tf.nn.softmax_cross_entropy_with_logits_v2
Below is outcome of before and after the optimization.

dimension of final output for vgg is [?, 3]
dimension of final output for yolo is [?, 1, 1, 3]

`softmax = tf.nn.softmax_cross_entropy_with_logits_v2(logits=vgg_pred, labels=y, dim=-1)`
print(softmax)

run 1
vgg output
[ 1110.82873535     0.          1372.21044922  1051.81994629  5117.62011719]
after optimization
[    0.          8890.65722656     0.             0.          6899.49121094]
yolo output
[[[       0. ]] [[ 8094592. ]] [[       0. ]] [[ 8089163.5]] [[ 8069554.5]]]
after optimization
**[[[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]]]**

run2
vgg output
[    0.             0.          2072.43017578     0.          2563.55004883]
after optimization
[  8293.29980469  11375.015625        0.          12365.3359375   1054.58935547]
yolo output
[[[ 8573584.]] [[       0.]] [[       0.]] [[ 8399384.]] [[ 9033760.]]]
after optimization
**[[[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]]]**

run3
vgg output
[    0.          2648.55541992     0.          2143.43066406     0.        ] 
after optimization
[ 12912.1875          0.          13529.96679688      0.          12147.125     ]
yolo output
[[[ 7737216.]] [[ 9160056.]] [[ 7647175.]] [[ 7012569.]] [[ 8301956.]]]
after optimization
**[[[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]]]**

run4
vgg output
[ 1804.08349609     0.             0.             0.          2595.27880859]
after optimization
[     0.          11371.57226562  10606.33691406  11048.16992188      0.        ]
yolo output
[[[ 8434419. ]] [[ 7721689.5]] [[       0. ]] [[ 8754676. ]] [[ 6599649.5]]]
after optimization
**[[[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]] [[ 1.09861231]]]**

**Describe the expected behavior**
I believe softmax outcome should be different in each run. It doesn't happen when I use tf.nn.softmax_cross_entropy_with_logits, which is to be removed in future version.

**Other info / logs**
Architecture of vgg and yolo is bit different. Both are manually editted by me. Training vgg gives good enough result; however, training yolo gives poor outcome. Particularly, error is not properly calculated, so the network is not trained at all.

**Comment by jvishnuvardhan** [comment]
@shkdidrlf Please post this kind of support questions at [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a big community to support. Github is mainly for addressing bugs in installation and performance. If you think it is related to a bug in the TF, then please provide a code to reproduce the bug so that we can find root cause of the issue. Thanks!
